src/cli/cli_coap.cpp

```diff
@@ -481,11 +481,18 @@ template <> otError Coap::Process<Cmd("get")>(Arg aArgs[]) { return ProcessReque
  *         * `con`: Confirmable
  *         * `non-con`: Non-confirmable (default)
  *         * `block-`: Use this option, followed by the block-wise value,
- *            to send blocks with a random payload. Valid
- *            values are: `block-16`, `block-32`, `block-64`, `block-128`,
+ *            to send blocks with a randomly generated number of bytes
+ *            for the payload. Valid values are:
+ *            `block-16`, `block-32`, `block-64`, `block-128`,
  *            `block-256`, `block-512`, or `block-1024`.
- *   * `payload`: CoAP payload request. If `type` is "block-",
- *     specify the amount of blocks to send using the `payload` option.
+ *   * `payload`: CoAP payload request, which if used is either a string or an
+ *     integer, depending on the `type`. If the `type` is `con` or `non-con`,
+ *     the `payload` parameter is optional. If you leave out the
+ *     `payload` parameter, an empty payload is sent. However, If you use the
+ *     `payload` parameter, its value must be a string, such as
+ *     `hellothere` in the example shown below. If `the type` is `block-`,
```

Comment:
Remove "in the example shown below", because if looking at this source, the example is before it.

It fine to just say "such as `hellothere`."